### PR TITLE
Masterbeta

### PR DESCRIPTION
**Pull request by JWeb56**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -207,6 +207,9 @@ dmypy.json
 ./data/
 src/python/psic/tests/**/output
 
+## John's stats test data
+src/python/psic/stats/data
+
```
Comment (Matmorcat) [inline .gitignore]: @JWeb56 Data should exist in the existing data folder. You can create a sub-folder called `stats` in the `data` folder and move it there

File: Pipfile [hunk 2]
```diff
@@ -23,3 +23,6 @@ jupyter = "*"
 pandas = "*"
 Pillow = "*"
 jsonpickle = "*"
+imageio = "*"
+matplotlib = "*"
+tensorflow = "*"
```
Comment (Matmorcat) [inline Pipfile]: @JWeb56 For whatever reason, trying to `pipenv lock` with `tensorflow = "*"` as a dependency causes an error!

File: src/python/psic/cnn_model/sample_model_with_training.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+for i, index in enumerate(np.random.choice(x_test.shape[0], size=15, replace=False)):
+    ax = figure.add_subplot(3, 5, i + 1, xticks=[], yticks=[])
+    # Display each image
+    ax.imshow(np.squeeze(x_test[index]))
+    predict_index = np.argmax(y_hat[index])
+    true_index = np.argmax(y_test[index])
+    # Set the title for each image
+    ax.set_title("{} ({})".format(fashion_mnist_labels[predict_index],
+                                  fashion_mnist_labels[true_index]),
+                                  color=("green" if predict_index == true_index else "red"))
```
Comment (Matmorcat) [inline src/python/psic/cnn_model/sample_model_with_training.py]: @JWeb56 I know that this is just a stub using the mnist intro, but be aware that there are many formatting errors listed by @pep8speaks in the comments of this document that need to be corrected before submitting to master!

File: src/python/psic/setup.py [hunk 4]
```diff
@@ -34,7 +34,8 @@
     ],
     keywords='python',
     license='MIT license',
-    install_requires=['jupyter', 'pandas', 'pillow', 'requests', 'tqdm'],
+    install_requires=['jupyter', 'pandas', 'pillow', 'requests', 'tqdm', 'imageio', 'imageio', 'matplotlib',
```
Comment (Matmorcat) [inline src/python/psic/setup.py]: `imageio` is duplicated and missing some packages

File: src/python/psic/stats/stats/bin/activate [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+pydoc () {
+    python -m pydoc "$@"
+}
+
+# This should detect bash and zsh, which have a hash command that must
+# be called to get it to forget past commands.  Without forgetting
+# past commands the $PATH changes we made may not be respected
+if [ -n "${BASH-}" ] || [ -n "${ZSH_VERSION-}" ] ; then
+    hash -r 2>/dev/null
+fi
```
Comment (Matmorcat) [inline src/python/psic/stats/stats/bin/activate]: Will need to work out how to fix plots not displaying. Using a script to create an environment is not ideal.

File: src/python/psic/setup.py (hunk 4 above)
Comment (Matmorcat) [reply src/python/psic/setup.py]: Fixed by #130

File: .gitignore (hunk 1 above)
Comment (JWeb56) [reply .gitignore]: According to what exactly?

File: src/python/psic/cnn_model/sample_model_with_training.py (hunk 3 above)
Comment (JWeb56) [reply src/python/psic/cnn_model/sample_model_with_training.py]: No, they don't need to be fixed to be committed to master. You WANT them to be fixed, but there are no errors whatsoever with this code.

File: src/python/psic/stats/stats/bin/activate (hunk 5 above)
Comment (JWeb56) [reply src/python/psic/stats/stats/bin/activate]: Ok

**Pull request closed without merging**